app/controllers/users_controller.rb

```diff
@@ -1,15 +1,33 @@
 class UsersController < ApplicationController
-  before_action :require_login, only: :show
+  before_action :require_login, only: [:show, :invite]
 
   def new
+    @token = params[:token]
+    redirect_to root_path if @token && !Invitation.find_by(token: @token)
+
+    @email = params[:email]
     @user = User.new
   end
 
   def create
+    invitation = Invitation.find_by(token: params[:token])
+
     @user = User.new(users_params)
```

Comment:
Be careful pluralizing `whateverS_params`. The convention as far as I know is that the first part is the model you are getting params first singular. Given that, `users_params` looks to me like an array of params for multiple users.